lib/samson/process_utils.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Samson
+  module ProcessUtils
+    ATTRIBUTES = ['pid', 'ppid', 'gid', 'pcpu', 'user', 'start', 'args'].freeze
+
+    class << self
+      def ps_list
+        whitelists = ENV.fetch('PROCESS_WHITELIST', [])
+        pipe = IO.popen("ps -eo #{ATTRIBUTES.join(',')}")
+        # Filter the whitelist processes
+        filtered_processes = pipe.readlines[1..-1].reject do |lines|
+          whitelists.any? { |key| lines.include?(key) }
+        end
+        filtered_processes.map do |line|
+          Hash[ATTRIBUTES.zip line.lstrip.split(/\s+/, ATTRIBUTES.size)]
+        end
+      end
+
+      def report_to_statsd
+        ps_list.each do |process|
+          timed_out = Time.now.to_i - Time.parse(process.fetch('start')).to_i
```

Comment:
prefer `Integer(value)` over `value.to_i` since it will fail when it is not an integer

Reply:
Time parse will not return nil, It may return ArgumentError, I have rescued it and will add a monitor for this date issues.